modules/auxiliary/gather/ldap_query.rb

```diff
@@ -314,22 +309,11 @@ def run
           entries = perform_ldap_query(ldap, filter, attributes)
           print_error("No entries could be found for #{datastore['QUERY_FILTER']}!") if entries.nil? || entries.empty?
         else
-          filter_string = nil
-          attributes = nil
-          @default_settings['queries'].each do |entry|
-            next unless entry['action'] == datastore['ACTION']
-
-            filter_string = entry['filter']
-            attributes = entry['attributes']
-            break
-          end
-
-          if attributes&.empty? || filter_string&.empty?
-            fail_with(Failure::BadConfig, "Couldn't find and/or load the attributes and filter string for #{datastore['ACTION']}. Check the validity of the YAML files at #{@default_settings_file_path} and #{@user_config_file_path}!")
-          end
+          query = @loaded_queries[datastore['ACTION']]
+          fail_with(Failure::BadConfig, "Invalid action: #{datastore['ACTION']}") unless query
 
-          filter = Net::LDAP::Filter.construct(filter_string)
-          entries = perform_ldap_query(ldap, filter, attributes)
+          filter = Net::LDAP::Filter.construct(query['filter'])
```

Comment:
This still needs to be wrapped in exception handling. The `Net::LDAP::Filter` unfortunately, whilst pretty good with most queries, doesn't handle everything as I encountered when trying to do some Exchange queries that are perfectly valid LDAP that for w.e reason `Net::LDAP::Filter` does just not want to accept. So yeah if you don't catch the exception that is thrown when this happens then you get shenanigans.